packages/nameguard-sdk/src/index.ts

```diff
@@ -793,22 +902,47 @@ export class NameGuard {
       options.body = JSON.stringify(body);
     }
 
-    const response = await fetch(url, options);
-
-    if (!response.ok) {
-      throw new NameGuardError(
-        response.status,
-        `Failed to perform request to ${path}.`,
-      );
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        throw new NameGuardError(
+          NameGuardErrorType.Server,
+          response.status,
+          `Server responded with status ${response.status}`,
+        );
+      }
+
+      return await response.json();
+    } catch (error: unknown) {
+      if (error instanceof NameGuardError) {
+        throw error;
+      } else if (error instanceof DOMException && error.name === "AbortError") {
+        throw new NameGuardError(NameGuardErrorType.Abort);
+      } else if (
+        error instanceof TypeError &&
+        error.message === "Failed to fetch"
+      ) {
+        throw new NameGuardError(NameGuardErrorType.Network);
+      } else {
+        throw new NameGuardError(
+          NameGuardErrorType.Unknown,
+          undefined,
+          isError(error) ? error.message : "Unknown error",
+        );
+      }
     }
-
-    return await response.json();
   }
 
   public abortAllRequests(): void {
     this.abortController.abort();
-
     this.abortController = new AbortController();
+
+    throw new NameGuardError(
```

Comment:
Not sure if this function should throw an error?

Reply:
At the time of spiking the PR I thought this could be slightly helpful. But I've reverted the changes and added a new doc block.